services/core-api/app/api/activity/resource/activities_mark_as_read.py

```diff
@@ -0,0 +1,22 @@
+from flask_restplus import Resource, reqparse
+
+from app.api.activity.models.activity_notification import ActivityNotification
+from app.api.utils.access_decorators import EDIT_DO, requires_any_of, MINESPACE_PROPONENT
+from app.api.utils.resources_mixins import UserMixin
+
+
+class ActivityMarkAsReadResource(Resource, UserMixin):
+    @requires_any_of([EDIT_DO, MINESPACE_PROPONENT])
+    def patch(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument(
+            'activity_guids',
+            type=list,
+            location='json',
+            required=True,
+            help='List of activity guids to mark as read')
+        args = parser.parse_args()
+        activity_guids = args.get('activity_guids')
+        for activity_guid in activity_guids:
+            activity = ActivityNotification.find_by_guid(activity_guid)
+            ActivityNotification.update(activity)
```

Comment:
We can probably add in an `mark_as_read_many` method on `ActivityNotification` with a query that would be something like this instead of looking up each individually

```python
ActivityNotification.filter(ActivityNotification.notification_guid.in_(notification_guids))
    .update({'notification_read': True})
```